doc/src/manual/environment-variables.md

```diff
@@ -0,0 +1,138 @@
+# Environment Variables in Julia
+
+The environment variables that Julia uses generally start with `JULIA`. If [`Base.versioninfo()`](https://github.com/JuliaLang/julia/blob/master/base/interactiveutil.jl#L247) is called with `verbose` equal to `true`, then the output will list all environment variables for which `JULIA` appears in the name.
+
+## File locations
+
+### `JULIA_HOME`
+The absolute path of the directory containing the Julia executable, which sets the global variable [`Base.JULIA_HOME`](https://github.com/JuliaLang/julia/blob/master/base/initdefs.jl#L90).  If `$JULIA_HOME` is not set, then `Base.JULIA_HOME` defaults to its compile-time value.
+
+The executable itself is one of 
+```
+$JULIA_HOME/julia
+$JULIA_HOME/julia-debug
+```
+by default.
+
+The global variable [`Base.DATAROOTDIR`](https://github.com/JuliaLang/julia/blob/master/base/Makefile#L62)  determines a relative path from `Base.JULIA_HOME` to the data directory associated with Julia; then the directory.  Then the path 
```

Comment:
The idea behind linking the function and variable names to locations in the source code is a good one, but the problem is that any edits to those files will change what's at the given line. It would get kind of messy for every PR to track the line numbers and update this file accordingly. So perhaps it would be better to link to documentation for the functions and/or variables if they're available, otherwise don't provide a link.